### ISSUE TITLE
Update trigger event types to match R1 data model

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/ctapipe#2094]
**Describe the bug**

The trigger event types definition in ctapipe.containers currently does not match the released R1 data model document.

Specifically, it distinguishes multiple different kinds of pedestal events, where the R1 data model only knows a single `PEDESTAL` event type.

**Additional context**

In simulations at least (and probably we also need this for observed data) these different kinds of pedestal events are known. Where should we store this information if there is no event type for it?

**Comment by kosack** [comment]
Can you give more information? What types of pedestal events are distinguished in simtel?

**Comment by maxnoe** [comment]
Basically the one we have currently in the enum:

```
    0: EventType.DARK_PEDESTAL,
    1: EventType.SKY_PEDESTAL,
    2: EventType.SINGLE_PE,
    3: EventType.FLATFIELD,
```

So it distinguishes between pedestals that are taken with closed or open shutter and between single_pe (internal illumination) and flatfield (external illumination).

**Comment by kosack** [comment]
Hmm.  that enum came directly from the data model originally (but perhaps an older version).  I didn't actually notice it was re-numbered or lost the "dark pedestal" type.  That may actually be a problem in the R1 Doc.

**Comment by kosack** [comment]
Ah, I see the change:

> Note that there are different types of pedestals: sky, dark, and electronic. The distinction between these pedestal types is handled at DPPS, based on monitoring data information transmitted to ACADA via the Monitoring system interface.

**Comment by kosack** [comment]
So I would say we can close this: there is no need to match the R1 model for event types, as it explicitly says we can make more detailed types at the DL1 (and maybe DL0) level with more information available at those levels.

The question may be if we keep a single event type (which exactly matches R0) and add a subtype, or if we simply re-define/re-map the event types to a more flexible model

**Comment by maxnoe** [comment]
@kosack ok, good. But we still need to adapt an important thing: these trigger types are defined per telescope event, not per array event.

E.g. a physics trigger event can have `LST_HARDWARE` for the LSTs and `STEREO` for MSTs and `STEREO_LONG` for long events in some MSTs.

**Comment by kosack** [comment]
Yes, those types are only per-telescope, so should probably go in the `TelescopeTriggerContainer` as the telescope trigger type.  Stereo events in observed data can only come from SWAT, which combines the stereo LST triggers and individual telescope triggers into a single subarray trigger.  So in the end, the only type that should be received there would be a "subarray stereo" type (and I think any info on the hardware trigger is only in the telescope trigger data). Have to confirm in the DL0 model, as that is where the SWAT info is really stored (at R1, we are still really only at telescope-level).

So I think we should just have different Enums for telescope trigger types and subarray trigger types.
